dotnet/src/Connectors/VectorData.Abstractions/VectorData.Abstractions.csproj

```diff
@@ -0,0 +1,38 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- THIS PROPERTY GROUP MUST COME FIRST -->
+    <AssemblyName>Microsoft.Extensions.VectorData.Abstractions</AssemblyName>
+    <RootNamespace>Microsoft.Extensions.VectorData</RootNamespace>
+    <TargetFrameworks>net8.0;netstandard2.0;net462</TargetFrameworks>
+    <EnablePackageValidation>true</EnablePackageValidation>
+  </PropertyGroup>
+
+<!--
+  TO BE ENABLED AFTER THE PACKAGE IS READY FOR PUBLISHING
+  <Import Project="$(RepoRoot)/dotnet/nuget/nuget-package.props" />
+
+  <PropertyGroup>
+    <PackageVersion>9.0.0-preview.9.24001.1</PackageVersion>
+    <Title>Semantic Kernel - Abstractions</Title>
+    <PackageId>$(AssemblyName)</PackageId>
+    <PackageDescription>Abstractions for vector database access.
+
+Commonly Used Types:
+Microsoft.Extensions.VectorData.IVectorStore
+Microsoft.Extensions.VectorData.IVectorStoreRecordCollection&lt;TKey, TRecord&gt;</PackageDescription>
+    <PackageReadmeFile>PACKAGE.md</PackageReadmeFile>
+    <Product></Product>
+    <PackageTags>Vector, Database, SDK</PackageTags>
+    <Description>$(PackageDescription)</Description>
+    <PackageProjectUrl>https://dot.net/</PackageProjectUrl>
+  </PropertyGroup>
+-->
+
+  <ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'net462' ">
+    <Compile Include="$(RepoRoot)/dotnet/src/InternalUtilities/src/Diagnostics/IsExternalInit.cs" Link="Diagnostics/IsExternalInit.cs" />
```

Comment:
Yeah, I was thinking that when we move it, we can just copy the file into the project or rely on whatever mechanism the folks in the other repo may be using.  I also already removed references to all optional internal utilities like Verify so that we minimize this problem.